internal/server/iigointernal/executive.go

```diff
@@ -0,0 +1,148 @@
+	result, _ := e.clientPresident.SetTaxationAmount(islandsResources)
+	return result
+}
+
+// broadcastTaxation broadcasts the tax amount decided by the president to all island still in the game.
+func (e *executive) broadcastTaxation(islandsResources map[shared.ClientID]shared.Resources) {
+	e.budget -= serviceCharge
+	taxAmountMap := e.getTaxMap(islandsResources)
+	for _, v := range getIslandAlive() {
+		d := baseclient.Communication{T: baseclient.CommunicationInt, IntegerData: int(taxAmountMap[shared.ClientID(int(v))])}
+		data := make(map[int]baseclient.Communication)
+		data[TaxAmount] = d
+		communicateWithIslands(shared.TeamIDs[int(v)], shared.TeamIDs[e.ID], data)
+	}
+}
+
+// Send Tax map all the remaining islands
+// Called by orchestration at the end of the turn
+func (e *executive) getAllocationRequests(commonPool shared.Resources) map[shared.ClientID]shared.Resources {
+	e.budget -= serviceCharge
+	result, _ := e.clientPresident.EvaluateAllocationRequests(e.ResourceRequests, commonPool)
+	return result
+}
+
+func (e *executive) requestAllocationRequest() {
+	allocRequests := make(map[shared.ClientID]shared.Resources)
+	for _, v := range getIslandAlive() {
+		allocRequests[shared.ClientID(int(v))] = iigoClients[shared.ClientID(int(v))].CommonPoolResourceRequest()
+	}
+	AllocationAmountMapExport = allocRequests
+	e.setAllocationRequest(allocRequests)
+
+}
+
+// replyAllocationRequest broadcasts the allocation of resources decided by the president
+// to all islands alive
+func (e *executive) replyAllocationRequest(commonPool shared.Resources) {
+	e.budget -= serviceCharge
+	allocationMap := e.getAllocationRequests(commonPool)
+	for _, v := range getIslandAlive() {
+		d := baseclient.Communication{T: baseclient.CommunicationInt, IntegerData: int(allocationMap[shared.ClientID(int(v))])}
+		data := make(map[int]baseclient.Communication)
+		data[AllocationAmount] = d
+		communicateWithIslands(shared.TeamIDs[int(v)], shared.TeamIDs[e.ID], data)
+	}
+}
+
+// appointNextSpeaker returns the island id of the island appointed to be speaker in the next turn.
+func (e *executive) appointNextSpeaker(clientIDs []shared.ClientID) shared.ClientID {
+	e.budget -= serviceCharge
+	var election voting.Election
+	election.ProposeElection(baseclient.Speaker, voting.Plurality)
+	election.OpenBallot(clientIDs)
+	election.Vote(iigoClients)
+	return election.CloseBallot()
+}
+
+// withdrawSpeakerSalary withdraws the salary for speaker from the common pool.
+func (e *executive) withdrawSpeakerSalary(gameState *gamestate.GameState) error {
+	var speakerSalary = shared.Resources(rules.VariableMap["speakerSalary"].Values[0])
+	var withdrawError = WithdrawFromCommonPool(speakerSalary, gameState)
+	if withdrawError != nil {
+		e.speakerSalary = speakerSalary
+	}
+	return withdrawError
+}
+
+// sendSpeakerSalary send speaker's salary to the speaker.
+func (e *executive) sendSpeakerSalary() {
+	amount, _ := e.clientPresident.PaySpeaker(e.speakerSalary)
```

Comment:
#127

Reply:
Same again. Any function call to do with client_insert role_ needs to give the islands the choice to not perform actions. This ahs not been implemented yet

Reply:
See above